instrumentation/apache-camel-2.20/src/main/java/io/opentelemetry/javaagent/instrumentation/apachecamel/decorators/HttpSpanDecorator.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+// Includes work from:
+/*
+ * Apache Camel Opentracing Component
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * <p>http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * <p>Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.javaagent.instrumentation.apachecamel.decorators;
+
+import io.grpc.Context;
+import io.opentelemetry.instrumentation.api.tracer.BaseTracer;
+import io.opentelemetry.javaagent.instrumentation.apachecamel.CamelDirection;
+import io.opentelemetry.trace.Span;
+import io.opentelemetry.trace.attributes.SemanticAttributes;
+import java.net.MalformedURLException;
+import java.net.URL;
+import org.apache.camel.Endpoint;
+import org.apache.camel.Exchange;
+
+class HttpSpanDecorator extends BaseSpanDecorator {
+
+  private static final String POST_METHOD = "POST";
+  private static final String GET_METHOD = "GET";
+
+  public static String getHttpMethod(Exchange exchange, Endpoint endpoint) {
+    // 1. Use method provided in header.
+    Object method = exchange.getIn().getHeader(Exchange.HTTP_METHOD);
+    if (method instanceof String) {
+      return (String) method;
+    }
+
+    // 2. GET if query string is provided in header.
+    if (exchange.getIn().getHeader(Exchange.HTTP_QUERY) != null) {
+      return GET_METHOD;
+    }
+
+    // 3. GET if endpoint is configured with a query string.
+    if (endpoint.getEndpointUri().indexOf('?') != -1) {
+      return GET_METHOD;
+    }
+
+    // 4. POST if there is data to send (body is not null).
+    if (exchange.getIn().getBody() != null) {
+      return POST_METHOD;
+    }
+
+    // 5. GET otherwise.
+    return GET_METHOD;
+  }
+
+  @Override
+  public String getOperationName(Exchange exchange, Endpoint endpoint) {
+    // Based on HTTP component documentation:
+    return getHttpMethod(exchange, endpoint);
```

Comment:
HTTP clients should use `HTTP METHOD` span name:

> HTTP client spans SHOULD be using conservative, low cardinality names formed from the available parameters of an HTTP request, such as "HTTP {METHOD_NAME}".

In case of inbound spans I think using the path would be fine here.

Reply:
Yep, we can do that, since we already update server span.